src/pathfinding_service/service.py

```diff
@@ -66,15 +69,18 @@ def __init__(  # pylint: disable=too-many-arguments
             user_deposit_contract_address=self.user_deposit_contract.address,
             allow_create=True,
         )
-        self.token_networks = self._load_token_networks()
 
+        self.matrix_listener = MatrixListener(
+            private_key=private_key,
+            chain_id=self.chain_id,
+            service_room_suffix=PATH_FINDING_BROADCASTING_ROOM,
+            message_received_callback=self.handle_message,
+            address_reachability_changed_callback=self.handle_reachability_change,
+        )
+
+        self.token_networks = self._load_token_networks()
         try:
-            self.matrix_listener = MatrixListener(
-                private_key=private_key,
-                chain_id=self.chain_id,
-                callback=self.handle_message,
-                service_room_suffix=PATH_FINDING_BROADCASTING_ROOM,
-            )
+            self.matrix_listener.start_client()
```

Comment:
Why do we need both `matrix_listener.start_client()` and `matrix_listener.start()`?

If we move `matrix_listener.start_client()` to `_run`, can we stop mocking the MatrixListener in all tests, since it does not cause any network connections if the PFS is instantiated without calling `_run`?

Reply:
Good point, fixed!